### PR TITLE
Validator fixes

### PR DESCRIPTION
**Pull request by raynaudoe**
**On this PR**:
- Fixed reading Tx length.
- Fixed read amino uint64.
- When a Vote gets rejected, vote's data (type, height, round) gets returned alongside with error code.
- Improved Zemu validator test with a combination of good/bad transaction types.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,15 +19,14 @@ enable_testing()
 
 cmake_policy(SET CMP0025 NEW)
 set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_VERBOSE_MAKEFILE ON)
```
Comment (jleni) [inline CMakeLists.txt:22]:
```suggestion

```
Let's desactivate this

File: app/src/common/actions.c [hunk 2]
```diff
@@ -43,8 +43,16 @@ void app_sign() {
...
-            io_exchange(CHANNEL_APDU | IO_RETURN_AFTER_TX, 2);
+            //Return the conflicting vote data
+            MEMZERO(G_io_apdu_buffer, IO_APDU_BUFFER_SIZE);
+            G_io_apdu_buffer[0] = vote.Type;
+            uint16_t offset = 1;
+            memcpy(&G_io_apdu_buffer[offset], &vote.Height, sizeof(vote.Height));
+            offset += sizeof(vote.Height);
+            memcpy(&G_io_apdu_buffer[offset], &vote.Round, sizeof(vote.Round));
+            offset += sizeof(vote.Round);
+            set_code(G_io_apdu_buffer, offset, APDU_CODE_COMMAND_NOT_ALLOWED);
```
Comment (jleni) [inline app/src/common/actions.c:54]: can we add here:
- vote type
- information about the current state too (round, height, type) ?

That way it is possible to understand the reason of the rejection

File: app/src/validator/parser_validator.c [hunk 3]
```diff
@@ -79,29 +80,23 @@ __Z_INLINE parser_error_t readRawVarint(parser_context_t *ctx, uint64_t *value)
...
-        shift += 7;
+    int64_t v = 0;
+    p += 7;
+    for (int8_t i = 0; i < 8; i++, p--) {
+        v <<= 8;
+        v += *p;
     }
 
-    ctx->lastConsumed += 8;
+    *value = v;
```
Comment (jleni) [inline app/src/validator/parser_validator.c:97]: we have a weird situation here. We are setting an unsigned uint64 with a signed int64 ?

File: app/src/validator/parser_validator.c [hunk 4]
```diff
@@ -79,29 +80,23 @@ __Z_INLINE parser_error_t readRawVarint(parser_context_t *ctx, uint64_t *value)
...
-        const uint64_t tmp = ((*p) & 0x7Fu);
-
-        if (shift == 63 && tmp > 1) {
-            return parser_value_out_of_range;
-        }
-
-        *value += tmp << shift;
-
-        shift += 7;
+    int64_t v = 0;
```
Comment (jleni) [inline app/src/validator/parser_validator.c:90]: The code seems different here... 
from what I read, in the old code... there are only 7 bits per byte that can be used..
and there is a termination condition in the higher order bit.

In the new code, it is just expecting 8 bytes.. like a fixed64.
can we check this?

File: CMakeLists.txt (hunk 1 above)
Comment (raynaudoe) [reply CMakeLists.txt:22]: done!

File: app/src/common/actions.c (hunk 2 above)
Comment (raynaudoe) [reply app/src/common/actions.c:54]: ok, will return the full [vote_state][current_vote][error_code]

**Pull request merged**